pkg/operator/sync.go

```diff
@@ -166,37 +169,121 @@ func (optr *Operator) syncWebhookConfiguration() error {
 }
 
 func (optr *Operator) syncValidatingWebhook() error {
+	client := optr.kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations()
+	ctx := context.TODO()
 	expected := mapiv1.NewValidatingWebhookConfiguration()
-	if webhook, updated, err := applyValidatingWebhookConfiguration(
-		optr.dynamicClient,
-		events.NewLoggingEventRecorder(optr.name),
-		expected,
-		expectedValidatingWebhooksConfiguration(expected.Name, optr.generations),
-	); err != nil {
-		return err
-	} else if updated {
-		setValidatingWebhooksConfigurationGeneration(&optr.generations, webhook)
+
+	current, err := client.Get(ctx, expected.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// Doesn't exist yet so create a fresh configuration
+		if _, err := client.Create(ctx, expected, metav1.CreateOptions{}); err != nil {
+			return fmt.Errorf("error creating ValidatingWebhookConfiguration: %v", err)
+		}
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("error getting ValidatingWebhookConfiguration: %v", err)
+	}
+
+	// The webhook already exists, so merge the existing fields with the desired fields
+	if err := mergo.Merge(expected, current); err != nil {
```

Comment:
Is it going to preserve custom labels we don't set on from the `current` resource? Are we ok with that? cc @michaelgugino

Reply:
Yes it will, custom labels and annotations or even finalizers should someone wish to add one, will be preserved. I think this is desirable. We should try to be nice and allow users to modify things if they so desire, just as long as it doesn't affect the operation of our webhooks. Which in this case, I don't think it will.